HACKING.md

```diff
@@ -0,0 +1,99 @@
+Barnabas project. You can build all Barnabas Docker images by calling 
+`make docker_build` from the root of the Barnabas repository. Or you can build 
+an individual Docker image by running `make docker_build` from the 
+subdirectories with their respective Dockerfiles - e.g. `kafka_base`, 
+`kafka_statefulsets` etc.
+
+The `docker_build` target will always build the images under the 
+`enmasseproject` organization. This is necessary in order to be able to reuse 
+the base image you might have just built without modifying all Dockerfiles.
+
+To configure the `docker_build` target you can set environment variables:
+* `DOCKER_ORG` should contain the Docker organization for tagging/pushing 
+  the images (default is your username)
+* `DOCKER_TAG` configured Docker tag (default is `latest`)
+
+### Tagging and pushing Docker images
+
+Target `docker_tag` can be used to tag the Docker images built by the 
+`docker_build` target. This target is automatically called by the `docker_push` 
+target and doesn't have to be called separately. The `docker_push` target will 
+login to the configured container repository and push the images.
+
+To configure the `docker_tag` and `docker_push` targets you can set following 
+environment variables:
+* `DOCKER_ORG` configures the Docker organization for tagging/pushing the 
+  images (defaults to the value of the `$USER` environment variable)
+* `DOCKER_TAG` configured Docker tag (default is `latest`)
+* `DOCKER_REGISTRY` configures the Docker registry where the image will 
+  be pushed (default is `docker.io`)
+
+## Building everything
+
+`make all` command can be used to triger all the tasks above - build the 
+Docker images, tag them and push them to the configured repository.
+
+## Pushing images to the cluster's Docker repo
+
+When developing locally you might want to push the docker images to the docker
+repository running in your local OpenShift cluster. This can be quicker than
+pushing to dockerhub and works even without a network connection.
+
+1. Make sure your cluster is running,
+
+        oc cluster up
+
+   and that you're logged in as `developer`,
+
+        oc whoami
+    
+   If you're not logged in as `developer`, then do so:
+   
+        oc login -u developer
+        
+2. Log in to the Docker repo running in the local cluster:
+
+        docker login -u myproject -p `oc whoami -t` 172.30.1.1:5000
+        
+   `172.30.1.1:5000` happens to be the IP and port of the docker registry
+   running in the cluster (see `oc get svc -n default`). 
+        
+3. Now run `make` to push the development images to that docker repo:
+
+        DOCKER_REGISTRY=172.30.1.1:5000 DOCKER_ORG=myproject make all
+        
+4. Finally, when creating the new app from the template you need to
+   specify the Docker repo, otherwise OpenShift will still try to pull 
+   the image from dockerhub:
+   
+        oc new-app barnabas -p IMAGE_REPO_NAME=172.30.1.1:5000/myproject
+
+
```

Comment:
The above procedure is ok whan using the local `oc cluster up` but should be slightly different when using `minishift` or `minikube`. Should we describe these ?

Reply:
I'm inclined not to, because they're going to be pretty similar, so it wouldn't add much, and I think knowing how to do it for one it enough for developers to figure out how to do it for minishift or minikube.